test/test.vimspec

```diff
@@ -1093,4 +1095,79 @@ Describe selection=exclusive
     End
 End
 
+Describe g:clever_f_mark_direct
+    Before
+        new
+        let g:clever_f_mark_direct = 1
+        call clever_f#reset()
+        highlight link CleverFDirect CleverFDefaultLabel
+        call AddLine('pOge huga Hiyo poyo')
+        let old_across_no_line = g:clever_f_across_no_line
+        let g:clever_f_across_no_line = 0
+    End
+
+    After
+        close!
+        let g:clever_f_mark_direct = 0
+        let g:clever_f_across_no_line = old_across_no_line
+    End
+
+    It should highlight target characters automatically
+        normal! gg0
+        normal f
+        Assert Equals(len(filter(getmatches(), 'v:val.group==#"CleverFDirect"')), 11)
+    End
+
+    It should remove target highlights
+        normal! gg0
+        normal fh
+        Assert Equals(len(filter(getmatches(), 'v:val.group==#"CleverFDirect"')), 0)
+    End
+
+    It should finish with no error
+        normal! gg$
+        normal fp
+        Assert Equals(len(filter(getmatches(), 'v:val.group==#"CleverFDirect"')), 0)
+    End
+
+    Describe <Esc>
+        It should remove target highlights
+            normal! gg0
+            execute 'normal' "f\<Esc>"
+            Assert Equals(len(filter(getmatches(), 'v:val.group==#"CleverFDirect"')), 0)
+        End
+    End
+
+    Describe g:clever_f_smart_case
+        Before
+            let g:clever_f_smart_case = 1
+        End
+
+        After
+            let g:clever_f_smart_case = 0
+        End
+
+        It should make less characters highlighted
+            normal! gg0
+            normal f
+            Assert Equals(len(filter(getmatches(), 'v:val.group==#"CleverFDirect"')), 10)
```

Comment:
Could you check the positions (columns) of marks are intended?

Reply:
OK